play-services-core/src/main/java/org/microg/gms/gcm/McsService.java

```diff
@@ -485,6 +487,25 @@ private void handleAppMessage(DataMessageStanza msg) {
             for (ResolveInfo resolveInfo : infos) {
                 logd("Target: " + resolveInfo);
                 Intent targetIntent = new Intent(intent);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    try {
+                        Field field = Context.class.getField("DEVICE_IDLE_CONTROLLER");
+                        IBinder binder = (IBinder) Class.forName("android.os.ServiceManager")
+                                .getMethod("getService", String.class).invoke(null, field.get(null));
+                        if (binder != null) {
+                            Object deviceIdleController = Class.forName("android.os.IDeviceIdleController$Stub")
+                                    .getMethod("asInterface", IBinder.class).invoke(null, binder);
+                            int userId = (int) UserHandle.class.getMethod("getUserId", int.class)
+                                    .invoke(null, getPackageManager().getApplicationInfo(msg.category, 0).uid);
+                            logd("Adding app " + msg.category + " for userId " + userId + " to the temp whitelist");
+                            deviceIdleController.getClass()
+                                    .getMethod("addPowerSaveTempWhitelistApp", String.class, long.class, int.class, String.class)
+                                    .invoke(deviceIdleController, resolveInfo.activityInfo.packageName, 10000, userId, "GCM Push");
```

Comment:
It's `resolveInfo.activityInfo.packageName` here and `msg.category` three lines above. Better be consistent.

Or, maybe even add a `String packageName = msg.category;` and use that instead to make it more clear what it is. Also `msg.category` should be the same as `resolveInfo.activityInfo.packageName` anyway...

Reply:
Good point.  Consistency would be important.  I made the change you suggested.

Reply:
I just noticed that `resolveInfo.activityInfo.packageName` is used [a few lines below here](https://github.com/microg/android_packages_apps_GmsCore/blob/ee1910acc42bcf5b88d6e1154ae3c160fe1fdc2d/play-services-core/src/main/java/org/microg/gms/gcm/McsService.java#L510).  Should I just make this line use the `String packageName` too?

If so, should it use `resolveInfo.activityInfo.packageName` instead of `msg.category`?  That might make more sense due to the scope.

Reply:
I guess for the last occurence, it might make sense to use `resolveInfo.activityInfo.packageName`, as it also uses `resolveInfo.activityInfo.name`.

If you move the `String packageName =` further up (in the first line of that method), you could replace all occurences of `msg.category` with `packageName`. Even better, as I bet it isn't super obvious that `msg.category` is defined as the package name.

Reply:
Ok, I left the last occurrence unchanged and I changed all occurrences of `msg.category` to `packageName`.